### DISCUSSION TITLE
manual override with custom device

### DISCUSSION THREAD

**Discussion by kobyrne12** [zwave-js/zwave-js#6016] [General] [answered]
Hi, 
is it possible to force a discovered device to use a particular custom device config file? if for example i want to write a custom config file for testing purposes only,how can i get that device to use it and then switch back to either the default config or another custom config after testing.
Also i have a device where the manufacturer is appearing as not defined so in that situation i would like to apply the correct config file to it no matter what manufacturer, product or type that is advertised. it might be a db stored one or a custom one but i want to define what config file to use. is this possible ?
thanks 
ken

**Comment by AlCalzone** [comment]
If you're using Z-Wave UI:

https://zwave-js.github.io/zwave-js-ui/#/device-support/custom-device-files

**Reply by kobyrne12** [reply]
Yes, I am aware of the method but maybe I am missing something crucial here, so apologies, but let’s say I have a device X and I want to make sure it uses custom file abc.json,  how do I ensure that happens? Do I need to make sure the manufacturer Id, product id and type id match?

**Reply by AlCalzone** [reply]
That and the firmware range. Also if the file imports others, you need to put the imported files in the correct place in that directory too.

**Reply by kobyrne12** [reply]
ok, that makes sense but i am still struggling with this, my device TZ55S is not returning the expected info, so in my json file stored in "store/config/tz55s.json" i have tried both 0xffff and 0x65535 as the manufactory id but it still does not pick up the correct configuration after a reboot of zwavejs ui and re-interview. 
here is the file i am using: 
```
{
	"manufacturer": "TKB Home",
	"manufacturerId": "0xffff",
	"label": "TZ55S",
	"description": "Single Paddle Wall Dimmer",
	"devices": [
		{
			"productType": "0x0003",
			"productId": "0x0004"
			"zwaveAllianceId": [1359, 1360, 1361, 1362]
		}
	],
	"firmwareVersion": {
		"min": "0.0",
		"max": "255.255"
	},
	"associations": {
		"1": {
			"label": "Group 1",
			"description": "Group 1 Control",
			"maxNodes": 5,
			"isLifeline": true
		}
	},
	"paramInformation": [
		{
			"#": "1",
			"label": "Ignore start level bit",
			"valueSize": 1,
			"defaultValue": 1,
			"allowManualEntry": false,
			"options": [
				{
					"label": "Don't ignore start level",
					"value": 0
				},
				{
					"label": "Ignore start level",
					"value": 1
				}
			]
		},
		{
			"#": "3",
			"label": "Night light",
			"description": "Indicates if LED is on when load is ON",
			"valueSize": 1,
			"defaultValue": 0,
			"allowManualEntry": false,
			"options": [
				{
					"label": "Night LED on",
					"value": 0
				},
				{
					"label": "Night LED off",
					"value": 1
				}
			]
		},
		{
			"#": "4",
			"label": "Invert switch",
			"valueSize": 1,
			"defaultValue": 0,
			"allowManualEntry": false,
			"options": [
				{
					"label": "ON command on TOP",
					"value": 0
				},
				{
					"label": "OFF command on TOP",
					"value": 1
				}
			]
		},
		{
			"#": "19",
			"label": "LED transmission indicator",
			"description": "LED flickering during transmission",
			"valueSize": 1,
			"defaultValue": 2,
			"allowManualEntry": false,
			"options": [
				{
					"label": "No flicker",
					"value": 0
				},
				{
					"label": "Flicker all time",
					"value": 1
				},
				{
					"label": "One second flicker",
					"value": 2
				}
			]
		}
	]
}
```
and here is the log of the interview: 
```
2023-07-17T08:11:35.562Z - INTERVIEW STAGE COMPLETED 
Arg 0:
ProtocolInfo
2023-07-17T08:11:35.707Z - INTERVIEW STAGE COMPLETED 
Arg 0:
NodeInfo
2023-07-17T08:11:35.887Z - VALUE ADDED 
Arg 0:
└─commandClassName: Manufacturer Specific
└─commandClass: 114
└─property: manufacturerId
└─endpoint: 0
└─newValue: 65535
└─propertyName: manufacturerId
2023-07-17T08:11:35.893Z - VALUE ADDED 
Arg 0:
└─commandClassName: Manufacturer Specific
└─commandClass: 114
└─property: productType
└─endpoint: 0
└─newValue: 3
└─propertyName: productType
2023-07-17T08:11:35.898Z - VALUE ADDED 
Arg 0:
└─commandClassName: Manufacturer Specific
└─commandClass: 114
└─property: productId
└─endpoint: 0
└─newValue: 4
└─propertyName: productId
2023-07-17T08:11:36.128Z - VALUE ADDED 
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: libraryType
└─endpoint: 0
└─newValue: 3
└─propertyName: libraryType
2023-07-17T08:11:36.134Z - VALUE ADDED 
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: protocolVersion
└─endpoint: 0
└─newValue: 4.5
└─propertyName: protocolVersion
2023-07-17T08:11:36.140Z - VALUE ADDED 
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: firmwareVersions
└─endpoint: 0
└─newValue: 2.2
└─propertyName: firmwareVersions
2023-07-17T08:11:36.145Z - VALUE ADDED 
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: hardwareVersion
└─endpoint: 0
└─newValue: 1
└─propertyName: hardwareVersion
2023-07-17T08:11:38.658Z - VALUE ADDED 
Arg 0:
└─commandClassName: Multilevel Switch
└─commandClass: 38
└─property: currentValue
└─endpoint: 0
└─newValue: 0
└─propertyName: currentValue
2023-07-17T08:11:40.127Z - INTERVIEW STAGE COMPLETED 
Arg 0:
CommandClasses
```

**Comment by AlCalzone** [comment, marked as answer]
you're missing a comma at the end of the productId line:
```
			"productId": "0x0004"
			"zwaveAllianceId": [1359, 1360, 1361, 1362]
```
which makes the device file have an invalid syntax that cannot be parsed.

FWIW, following our [guide for writing config files](https://zwave-js.github.io/node-zwave-js/#/config-files/contributing-files?id=editor-setup) and using a proper IDE should highlight the error - not sure what you're using:
![grafik](https://github.com/zwave-js/node-zwave-js/assets/17641229/15638773-1a00-4e75-b007-98354346ca9e)

**Reply by kobyrne12** [reply]
perfect, totally missed that and when the comma was there i was using the wrong manufacture id. With the updated file it appears to have matched with that device now and all is well. thanks a million for the help, i am migrating over from smartthings where i had custom device handlers that i could associated with a device easily so i was really questioning my decision use HA. thanks again.
